os/fs/smartfs/smartfs_smart.c

```diff
@@ -1696,13 +1716,25 @@ static int smartfs_mkdir(struct inode *mountpt, const char *relpath, mode_t mode
 
 		/* Create the directory */
 
-		ret = smartfs_createentry(fs, parentdirsector, filename,
-								  SMARTFS_DIRENT_TYPE_DIR, mode, &entry, 0xFFFF, NULL);
-		if (ret != OK) {
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+		ret = smartfs_find_firstinvalidentry(fs, parentdirsector, &invalidentry);
+		if (ret == OK) {
+			invalidentry.firstsector = SMARTFS_ERASEDSTATE_16BIT;
+			ret = smartfs_replaceentry(fs, invalidentry, filename, SMARTFS_DIRENT_TYPE_DIR, mode, &entry, NULL);
+			if (ret != OK) {
+				goto errout_with_semaphore;
+			}
+		} else if (ret < 0 && ret != -ENOENT) {
```

Comment:
fs always return OK or errno, so below is correct.
```
if (ret == ok) {
   blah_blah;
} else {
   if (ret != -ENOENT) {
      bah bah!
   } else {
      lol lol!
   }
}

Reply:
Done